src/main/java/graphql/annotations/GraphQLAnnotations.java

```diff
@@ -554,8 +538,24 @@ protected GraphQLFieldDefinition getField(Method method) throws GraphQLAnnotatio
                     graphql.schema.GraphQLType graphQLType = finalTypeFunction.buildType(t, parameter.getAnnotatedType());
                     if (graphQLType instanceof GraphQLObjectType) {
                         GraphQLInputObjectType inputObject = getInputObject((GraphQLObjectType) graphQLType, "input");
+                        inputObject = (GraphQLInputObjectType) getOrPutInRegistry(inputObject);
                         graphQLType = inputObject;
+                    } else if (graphQLType instanceof GraphQLList) {
+                        Stack<graphql.schema.GraphQLType> typesStack= new Stack<>();
+                        typesStack = getWrappedTypesStack(graphQLType, typesStack);
+                        graphql.schema.GraphQLType wrappedType = typesStack.pop();
+                        if (wrappedType instanceof GraphQLObjectType){
```

Comment:
What if I have an interface? will this still work? (I don't remember if an interface is also an objectType)

Reply:
It won't, but Im not sure if we should handle it the same way